src/lib/y2network/autoinst/interfaces_reader.rb

```diff
@@ -0,0 +1,150 @@
+        if config.bootproto == BootProtocol::STATIC
+          # TODO: report if ipaddr missing for static config
+          ipaddr = IPAddress.from_string(interface_section.ipaddr)
+          # Assign first netmask, as prefixlen has precedence so it will overwrite it
+          ipaddr.netmask = interface_section.netmask if interface_section.netmask
+          ipaddr.prefix = interface_section.prefixlen.to_i if interface_section.prefixlen
+          broadcast = interface_section.broadcast && IPAddress.new(interface_section.broadcast)
+          remote = interface_section.remote_ipaddr && IPAddress.new(interface_section.remote_ipaddr)
+          config.ip = IPConfig.new(ipaddr, broadcast: broadcast, remote_address: remote)
+        end
+
+        config.startmode = Startmode.create(interface_section.startmode) if interface_section.startmode
+        config.startmode.priority = interface_section.ifplugd_priority if config.startmode.name == "ifplugd" && interface_section.ifplugd_priority
+        config.mtu = interface_section.mtu.to_i if interface_section.mtu
+        config.ethtool_options = interface_section.ethtool_options if interface_section.ethtool_options
+        config.firewall_zone = interface_section.zone if interface_section.zone
+      end
+
+      def load_wireless(config, interface_section)
+        config.mode = interface_section.wireless_mode if interface_section.wireless_mode
+        config.ap = interface_section.wireless_ap if interface_section.wireless_ap
+        config.bitrate = interface_section.wireless_bitrate.to_f if interface_section.wireless_bitrate
+        config.ca_cert = interface_section.wireless_ca_cert if interface_section.wireless_ca_cert
+        config.channel = interface_section.wireless_channel.to_i if interface_section.wireless_channel
+        config.client_cert = interface_section.wireless_client_cert if interface_section.wireless_client_cert
+        config.client_key = interface_section.wireless_client_key if interface_section.wireless_client_key
+        config.essid = interface_section.wireless_essid if interface_section.wireless_essid
+        config.auth_mode = interface_section.wireless_auth_mode.to_sym if interface_section.wireless_auth_mode
+        config.nick = interface_section.wireless_nick if interface_section.wireless_nick
+        config.nwid = interface_section.wireless_nwid if interface_section.wireless_nwid
+        config.wpa_anonymous_identity = interface_section.wireless_wpa_anonid if interface_section.wireless_wpa_anonid
+        config.wpa_identity = interface_section.wireless_wpa_identity if interface_section.wireless_wpa_identity
+        config.wpa_password = interface_section.wireless_wpa_password if interface_section.wireless_wpa_password
+        config.wpa_psk = interface_section.wireless_wpa_psk if interface_section.wireless_wpa_psk
+        config.keys = []
+        (0..3).each do |i|
+          key = interface_section.public_send(:"wireless_key#{i}")
```

Comment:
The section object could have a `#wireless_keys` method that abstracts these logic details.

Reply:
you mean as helper? makes sense.